### ISSUE TITLE
VillagerWaypointsTraverser Crashes game

### ISSUE THREAD

**Issue by AreYSerious** [G3rste/vsvillage#36]
just randomly staying near a village this happens.
Not sure how to replicate yet. will update if I find a way.

```
Running on 64 bit Windows 10.0.22631.0 with 32705 MB RAM
Game Version: v1.19.7 (Stable)
03.07.2024 19:24:22: Critical error occurred
Loaded Mods: Vanilla_PlusWorldGen@1.3.5, game@1.19.7, zoombuttonreborn@1.8.0, blocksoverlay@3.2.2, creative@1.19.7, survival@1.19.7, vsvillage@1.0.1, vsvillageaged@0.0.1, vsvillageindustrial@0.0.1, vsvillagetowers@0.0.1
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.Client.NoObf.SystemRenderEntities.OnBeforeRender(Single dt) in VintagestoryLib\Client\Systems\Render\RenderEntities.cs:line 49
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 186
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\ClientMain.cs:line 809
   at Vintagestory.Client.NoObf.ClientMain.MainRenderLoop(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 815
   at Vintagestory.Client.NoObf.ClientMain.MainGameLoop(Single deltaTime) in VintagestoryLib\Client\ClientMain.cs:line 731
   at Vintagestory.Client.GuiScreenRunningGame.RenderToPrimary(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 200
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 676
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 651
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 88
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 324
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass9_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 128
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93

Event Log entries for Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 28.06.2024 17:59:14, Site = , Source = Application Error, Message = Name der fehlerhaften Anwendung: Vintagestory.exe, Version: 1.19.7.0, Zeitstempel: 0x65310000
Name des fehlerhaften Moduls: coreclr.dll, Version: 7.0.1023.36312, Zeitstempel: 0x64b06d6c
Ausnahmecode: 0xc0000005
Fehleroffset: 0x00000000001c98a6
ID des fehlerhaften Prozesses: 0x0x5cd8
Startzeit der fehlerhaften Anwendung: 0x0x1dac9741020d23b
Pfad der fehlerhaften Anwendung: S:\VintageStoryEXP\Vintagestory\Vintagestory.exe
Pfad des fehlerhaften Moduls: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.10\coreclr.dll
Berichtskennung: 82c9cc2a-42b7-4939-832e-f7aa3c8875bc
Vollständiger Name des fehlerhaften Pakets: 
Anwendungs-ID, die relativ zum fehlerhaften Paket ist:  }
--------------
{ TimeGenerated = 27.06.2024 14:11:41, Site = , Source = Application Error, Message = Name der fehlerhaften Anwendung: Vintagestory.exe, Version: 1.19.7.0, Zeitstempel: 0x65310000
Name des fehlerhaften Moduls: coreclr.dll, Version: 7.0.1023.36312, Zeitstempel: 0x64b06d6c
Ausnahmecode: 0xc0000005
Fehleroffset: 0x00000000001c98a6
ID des fehlerhaften Prozesses: 0x0x6bf4
Startzeit der fehlerhaften Anwendung: 0x0x1dac889aebc9648
Pfad der fehlerhaften Anwendung: S:\VintageStoryEXP\Vintagestory\Vintagestory.exe
Pfad des fehlerhaften Moduls: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.10\coreclr.dll
Berichtskennung: 8a3ebe40-ec10-4474-80d4-42eee8137b6c
Vollständiger Name des fehlerhaften Pakets: 
Anwendungs-ID, die relativ zum fehlerhaften Paket ist:  }
--------------
{ TimeGenerated = 27.06.2024 13:08:17, Site = , Source = Application Error, Message = Name der fehlerhaften Anwendung: Vintagestory.exe, Version: 1.19.7.0, Zeitstempel: 0x65310000
Name des fehlerhaften Moduls: coreclr.dll, Version: 7.0.1023.36312, Zeitstempel: 0x64b06d6c
Ausnahmecode: 0xc0000005
Fehleroffset: 0x00000000001c98a6
ID des fehlerhaften Prozesses: 0x0x310c
Startzeit der fehlerhaften Anwendung: 0x0x1dac88246217a31
Pfad der fehlerhaften Anwendung: S:\VintageStoryEXP\Vintagestory\Vintagestory.exe
Pfad des fehlerhaften Moduls: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.10\coreclr.dll
Berichtskennung: 1bfeff7b-3617-4026-81e5-2979900e7a86
Vollständiger Name des fehlerhaften Pakets: 
Anwendungs-ID, die relativ zum fehlerhaften Paket ist:  }
```

[server-main.zip](https://github.com/user-attachments/files/16088577/server-main.zip)

**Comment by AreYSerious** [comment]
[server-main.zip](https://github.com/user-attachments/files/16088647/server-main.zip)
Also sometimes do not crash the game.

**Comment by G3rste** [comment]
Fixed with [v1.0.2](https://github.com/G3rste/vsvillage/pull/37)
